ci-operator/step-registry/upi/install/heterogeneous/powervs/upi-install-heterogeneous-powervs-commands.sh

```diff
@@ -0,0 +1,203 @@
+#!/bin/bash
+
+set -o nounset
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+if [ "${ADDITIONAL_WORKERS}" == "0" ]; then
+    echo "No additional workers requested"
+    exit 0
+fi
+
+function get_ready_nodes_count() {
+  oc get nodes \
+    -o jsonpath='{range .items[*]}{.metadata.name}{","}{.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}' | \
+    grep -c -E ",True$"
+}
+
+# wait_for_nodes_readiness loops until the number of ready nodes objects is equal to the desired one
+function wait_for_nodes_readiness()
+{
+  local expected_nodes=${1}
+  local max_retries=${2:-10}
+  local period=${3:-5}
+  for i in $(seq 1 "${max_retries}") max; do
+    if [ "${i}" == "max" ]; then
+      echo "[ERROR] Timeout reached. ${expected_nodes} ready nodes expected, found ${ready_nodes}... Failing."
+      return 1
+    fi
+    sleep "${period}m"
+    ready_nodes=$(get_ready_nodes_count)
+    if [ x"${ready_nodes}" == x"${expected_nodes}" ]; then
+        echo "[INFO] Found ${ready_nodes}/${expected_nodes} ready nodes, continuing..."
+        return 0
+    fi
+    echo "[INFO] - ${expected_nodes} ready nodes expected, found ${ready_nodes}..." \
+      "Waiting ${period}min before retrying (timeout in $(( (max_retries - i) * (period) ))min)..."
+  done
+}
+
+EXPECTED_NODES=$(( $(get_ready_nodes_count) + ADDITIONAL_WORKERS ))
+
+echo "Cluster type is ${CLUSTER_TYPE}"
+
+case "$CLUSTER_TYPE" in
+*ibmcloud*)
+  # Add code for ppc64le
+  if [ "${ADDITIONAL_WORKER_ARCHITECTURE}" == "ppc64le" ]; then
+      echo "Adding additional ppc64le nodes"
+      REGION="${LEASED_RESOURCE}"
+      IBMCLOUD_HOME_FOLDER=/tmp/ibmcloud
+      SERVICE_NAME=power-iaas
+      SERVICE_PLAN_NAME=power-virtual-server-group
+      WORKSPACE_NAME=rdr-mac-${REGION}-n1
+
+      PATH=${PATH}:/tmp
+      mkdir -p ${IBMCLOUD_HOME_FOLDER}
+      if [ -z "$(command -v ibmcloud)" ]; then
+        echo "ibmcloud CLI doesn't exist, installing"
+        curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+      fi
+      
+      function ic() {
+        HOME=${IBMCLOUD_HOME_FOLDER} ibmcloud "$@"
+      }
+
+      # Check if jq,yq,git and openshift-install are installed
+      if [ -z "$(command -v yq)" ]; then
+      	echo "yq is not installed, proceed to installing yq"
+      	curl -L "https://github.com/mikefarah/yq/releases/download/v4.30.5/yq_linux_$(uname -m | sed 's/aarch64/arm64/;s/x86_64/amd64/')" \
```

Comment:
Another version # to pull into the config if possible.

Reply:
Fix is in next commit.